### PR TITLE
Dev/0.5.0

### PR DESCRIPTION
**Pull request by adatzer**

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -1293,36 +1300,39 @@ const merge = (args) => {
...
-    return data;
-  }
-  const base64Enc = toBase64(data);
-  const urlBase64Enc = replaceAll(
-    replaceAll(replaceAll(base64Enc, '=', ''), '+', '-'),
-    '/',
-    '_'
-  );
-  return urlBase64Enc;
+  const rexp = createRegex('=|\\+|/', 'g');
```
Comment (jethron) [inline template.tpl]: Nit: `createRegex('[=+/]', 'g');`

File: template.tpl [hunk 2]
```diff
@@ -1195,12 +1196,12 @@ const spDefaultCustomDefs = {
...
 
-/*
+/**
  * Assumes logType argument is string.
  * Determines if logging is enabled.
  *
- * @param logType {string} - the logType set ('no', 'debug', 'always')
- * @returns - whether logging is enabled (boolean)
+ * @param {string} logType - The logType set ('no', 'debug', 'always')
+ * @returns {boolean} - Whether logging is enabled
```
Comment (jethron) [inline template.tpl]: Nit: Technically the `- ` prefix for descriptions is only valid for `@param` tags, not for `@returns`.

It's weird that the TypeScript examples use it for returns as well.

File: template.tpl [hunk 3]
```diff
@@ -1879,10 +1944,15 @@ const addContextEntities = (spEvent, evObj, tagConfig) => {
...
-/*
+/**
  * Given the client event object,
  *  - gets the Snowplow event from 'x-sp-tp2' (by Snowplow client)
  *  - or makes a Snowplow event
+ *
+ * @param {Object} evObj - The common event
+ * @param {Object} tagConfig - The tag configuration
+ * @param {function} definerFunction - The function to create the event definitions
+ * @param {Object} - The Snowplow event
```
Comment (jethron) [inline template.tpl]: Nit: missing `@returns`

File: template.tpl [hunk 4]
```diff
@@ -1293,36 +1300,39 @@ const merge = (args) => {
...
+        return match;
+    }
+  });
 };
 
-/*
+/**
  * Parses the cookie attributes of set-cookie header and returns the corresponding object.
  * e.g. parseCookie('name=value; Secure; HttpOnly') returns
  *       { name:value, Secure:true, HttpOnly:true }
```
Comment (jethron) [inline template.tpl]: Nit: [`@example`](https://jsdoc.app/tags-example.html)

File: template.tpl [hunk 5]
```diff
@@ -1587,11 +1632,11 @@ const getFromPath = (path, obj) => {
  *       setFromPath('a.0',   4, {a: {b: 0}}) => {a: [4]}
  *       setFromPath('a.2',   5, {a: [1,1,1]}) => {a: [1,1,5]}
  *
- * @param path {string | array} - the string to replace into
- * @param val {string} - the substring to replace
- * @param obj {Object} - the object to mutate
- * @param target {Object} - (optional) the object that the path refers to
- * @returns - the object mutated with the value set
+ * @param {(string|string[])} - The path where to set the value
```
Comment (jethron) [inline template.tpl]: Nit: Inconsistent missing param name.

File: template.tpl [hunk 6]
```diff
@@ -896,6 +896,7 @@ ___TEMPLATE_PARAMETERS___
 ___SANDBOXED_JS_FOR_SERVER___
 
 // Server-side tagging API imports
+const createRegex = require('createRegex');
```
Comment (jethron) [inline template.tpl:899]: Probably again need a note in the readme that v0.5.0 onwards requires v2 of the docker image or later.

**Review by jethron — APPROVED** [review]
LGTM!

**Comment by adatzer** [comment]
Thank you for reviewing @jethron !

**Pull request merged**